### ISSUE TITLE
Language server crashes when using import.meta.url

### ISSUE THREAD

**Issue by Soremwar** [denoland/deno#8864]
**Describe the bug**

The language server/extension crashes when using the import.meta.url variable.

**To Reproduce**

Just create any relative route with the URL api and see the magic happen.

**Video**
https://user-images.githubusercontent.com/42647963/102919956-ca313c80-4457-11eb-926d-0015827f386c.mp4

**Versions**

vscode: 1.52.0
deno: 1.6.2 (It can be reproduced on 1.6 and up)
extension: Canary 0.0.6

**Comment by kitsonk** [comment]
So what is happening is that `tsc` is complaining about the module not being a module, which technically it isn't at this particular point.

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error: Debug Failure. False expression: Containing file should be a module.
    at checkImportMetaProperty (deno:cli/tsc/00_typescript.js:69680:22)
    at checkMetaProperty (deno:cli/tsc/00_typescript.js:69655:24)
    at checkExpressionWorker (deno:cli/tsc/00_typescript.js:71979:28)
    at checkExpression (deno:cli/tsc/00_typescript.js:71870:38)
    at checkExpressionWithContextualType (deno:cli/tsc/00_typescript.js:71517:28)
    at getSignatureApplicabilityError (deno:cli/tsc/00_typescript.js:68085:35)
    at chooseOverload (deno:cli/tsc/00_typescript.js:68590:25)
    at resolveCall (deno:cli/tsc/00_typescript.js:68464:26)
    at resolveNewExpression (deno:cli/tsc/00_typescript.js:68929:24)
    at resolveSignature (deno:cli/tsc/00_typescript.js:69241:28)', cli/lsp/language_server.rs:501:38
```

If for example, this is used, nothing goes wrong:

```js
export {};

const url = new URL("a.js", import.meta.url);
```

I think we are just going to have to handle the failure a bit better, but it might not fix everything, as it is really hard to get `tsc` to think things are modules when they don't look like modules, sadly.

**Comment by cref** [comment]
I'm still getting this error.
See the output below.
The error can be triggered by typing `import.` in VS Code.
Can the issue be reopened?
```
Starting Deno language server...
  version: 1.12.0+1ad6575 (release, x86_64-pc-windows-msvc)
  executable: C:\Users\Michiel\.deno\bin\deno.EXE
Connected to "Visual Studio Code" 1.58.1
language server initialized
Server ready.
Unable to get completion info from TypeScript: Error: Debug Failure.
    at getTypeOfVariableOrParameterOrPropertyWorker (deno:cli/tsc/00_typescript.js:52849:22)
    at getTypeOfVariableOrParameterOrProperty (deno:cli/tsc/00_typescript.js:52813:28)
    at getTypeOfSymbol (deno:cli/tsc/00_typescript.js:53184:24)
    at getTypeOfSymbolAtLocation (deno:cli/tsc/00_typescript.js:66947:20)
    at Object.getTypeOfSymbolAtLocation (deno:cli/tsc/00_typescript.js:45177:35)
    at Object.getSymbolDisplayPartsDocumentationAndSymbolKind (deno:cli/tsc/00_typescript.js:135084:97)
    at deno:cli/tsc/00_typescript.js:124223:41
    at Object.runWithCancellationToken (deno:cli/tsc/00_typescript.js:45462:28)
    at createCompletionDetailsForSymbol (deno:cli/tsc/00_typescript.js:124222:30)
    at Object.getCompletionEntryDetails (deno:cli/tsc/00_typescript.js:124204:28)
[Error - 13:25:17] Request completionItem/resolve failed.
  Message: Internal error
  Code: -32603 
ERROR TSLS = Error: Debug Failure. False expression: Containing file should be a module.
    at checkImportMetaProperty (deno:cli/tsc/00_typescript.js:72543:22)
    at checkMetaProperty (deno:cli/tsc/00_typescript.js:72518:24)
    at checkExpressionWorker (deno:cli/tsc/00_typescript.js:74917:28)
    at checkExpression (deno:cli/tsc/00_typescript.js:74808:38)
    at checkExpressionStatement (deno:cli/tsc/00_typescript.js:77392:13)
    at checkSourceElementWorker (deno:cli/tsc/00_typescript.js:80162:28)
    at checkSourceElement (deno:cli/tsc/00_typescript.js:80036:17)
    at Object.forEach (deno:cli/tsc/00_typescript.js:384:30)
    at checkSourceFileWorker (deno:cli/tsc/00_typescript.js:80384:20)
    at checkSourceFile (deno:cli/tsc/00_typescript.js:80351:13)
```

**Comment by cref** [comment]
Looks like this is the same problem as #10641

**Comment by kitsonk** [comment]
Correct, this issue is closed, #10641 remains open.  The error posted is a duplicate of #10641.
